### PR TITLE
Added general referral event for caseworkers

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-872

### Change description ###

- Added general referral event and configured tab to display general referral fields.
- Removed clearing of general email object which will make it aligned with divorce 1.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            .description("General referral")
+            .explicitGrants()
+            .showSummary()
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .grant(CREATE_READ_UPDATE, CASEWORKER_COURTADMIN_CTSC)
+            .grant(READ, CASEWORKER_COURTADMIN_RDU, CASEWORKER_SUPERUSER, CASEWORKER_LEGAL_ADVISOR, SOLICITOR, CITIZEN))
+            .page("generalReferral")
+            .pageLabel("General referral")
+            .complex(CaseData::getGeneralReferral)
+            .mandatory(GeneralReferral::getGeneralReferralReason)
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java:53]: formatting

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_GENERAL_REFERRAL)
+            .forAllStates()
+            .name("General referral")
+            .description("General referral")
+            .explicitGrants()
+            .showSummary()
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java:46]: .showSummary(false)?

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        final CaseDetails<CaseData, State> beforeDetails
+    ) {
+        log.info("Caseworker general referral about to submit callback invoked");
+
+        var caseDataCopy = details.getData().toBuilder().build();
+
+        State endState;
+
+        if (caseDataCopy.getGeneralReferral().getGeneralReferralFeeRequired().toBoolean()) {
+            endState = AwaitingGeneralReferralPayment;
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java:74]: maybe a ternary assignment

**Review by SomeshRA — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java (hunk 2 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralReferral.java:46]: yes you are right summary is not shown in old divorce also

**Review by linusnorton — APPROVED** [review]

**Pull request merged**